pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -927,8 +927,17 @@ class Timestamp(_Timestamp):
             _tzinfo = tzinfo
 
         # reconstruct & check bounds
-        ts_input = datetime(dts.year, dts.month, dts.day, dts.hour, dts.min,
-                            dts.sec, dts.us, tzinfo=_tzinfo)
+        if _tzinfo is not None and treat_tz_as_pytz(_tzinfo):
+            # be careful about DST transition, #18319
```

Comment:
make an informative comments.

why woudn't you always localize?

Reply:
`tzinfo.localize` changing the `tzinfo` object is specific to pytz isn't it?